### ISSUE TITLE
Compatibility problem with multimedia filter

### ISSUE THREAD

**Issue by zabellemotte** [marcusgreen/moodle-qtype_wordselect#50]
This question type brokes the multimedia filter that is commonly used to integrate medias in questions.
To reproduce the bug :
- activate multimedia filter in contents 
   Administration > Site administration > Plugins > Filters > Manage filters
- create a wordselect question
- place some market words as question type supposes to
- use the link button to add a link towards a loaded mp3 file for example
- save and preview

In other question types, the edition forme shows a link toward the file.
In word select edition form, the multimedia filter seems to be loaded.
In other question types, the preview shows a player for the audio file that plays the audio.
In word select, the preview shows the same player but the audio does not play and an unending loading cercle appears.

**Comment by zabellemotte** [comment]
In Moodle 4.1, the wordselect question type seems to swap the multimedia effect filter and shows a link toward the audio file, as well in edition form as in preview.

**Comment by marcusgreen** [comment]
Thank you for your detailed report. I have an idea this is a bug that may happen with any filter. I have an automated test for the multilanguage filter, which you can see here
https://github.com/marcusgreen/moodle-qtype_wordselect/blob/master/tests/behat/filter_test.feature
But I suspect it depends on where the item is in the text. I will investigate, though I think a will not be trivial.

**Comment by marcusgreen** [comment]
I am about to release a verion for Moodle 4.5 and I want to confirm that this is a known issue that I do not have a work around for at the moment and am unlikely have a fix in the near future. My apologies.

**Comment by OT-AndreyUskov** [comment]
Hello, @marcusgreen.

Even if the media filter is disabled, the video and audio will be broken in the 'questiontext' area if you upload the video and audio not as links but using "Insert or edit an audio/video file" in Atto.

Errors in browser console:
GET https://LMS.DOMAIN/pluginfile.php/4358/qtype_wordselect/questiontext/351/1/3354/SampleVideo_1280x720_5mb.mp4?time=1739797006754 net::ERR_UNEXPECTED_PROXY_AUTH
GET https://LMS.DOMAIN/pluginfile.php/4358/qtype_wordselect/questiontext/351/1/3354/SampleAudio_0.7mb.mp3?time=1739797028903 net::ERR_UNEXPECTED_PROXY_AUTH

In the 'introduction' area, video and audio work without problems.

**Comment by marcusgreen** [comment]
Yes, the handling of filters and media uploads is broken, especially if it is placed after the first word (it might even be broken if it is before that). I think that pasting in as base64 is probably OK. It will take 2 to 4 days of development to fix. There is a related issue here
https://github.com/marcusgreen/moodle-qtype_wordselect/issues/42
I would like to get it fixed especially for the purposes of the mlang2 filter, but I cannot allocate the time required for the forseeable future. I should put a note in the plugins database to indicate there is this problem.
